pkg/utils/validator/clusterresourceplacement.go

```diff
@@ -346,3 +362,96 @@ func validateRolloutStrategy(rolloutStrategy placementv1beta1.RolloutStrategy) e
 
 	return apiErrors.NewAggregate(allErr)
 }
+
+// validatePropertySelector validates the property selector
+func validatePropertySelector(propertySelector *placementv1beta1.PropertySelector) error {
+	return validatePropertySelectorRequirements(propertySelector.MatchExpressions)
+}
+
+func validatePropertySelectorRequirements(propertySelectorRequirements []placementv1beta1.PropertySelectorRequirement) error {
+	var allErr []error
+	for i, req := range propertySelectorRequirements {
+		if err := validateName(req.Name); err != nil {
+			allErr = append(allErr, fmt.Errorf("invalid property name %s: %w", req.Name, err))
+		}
+		if err := validateOperator(req.Operator, req.Values); err != nil {
+			allErr = append(allErr, err)
+		}
+		if err := validateValues(req.Values); err != nil {
+			allErr = append(allErr, fmt.Errorf("invalid values for property %s: %w", req.Name, err))
+		}
+		if err := checkContradictions(i, req, propertySelectorRequirements); err != nil {
+			allErr = append(allErr, err)
+		}
+	}
+	if len(allErr) > 0 {
+		return apiErrors.NewAggregate(allErr)
+	}
+	return nil
+}
+
+func validateName(name string) error {
+	if err := validation.IsQualifiedName(name); err != nil {
+		return fmt.Errorf("name is not a valid Kubernetes label name: %v", err)
+	}
+	return nil
+}
+
+func validateOperator(op placementv1beta1.PropertySelectorOperator, values []string) error {
+	validOperators := map[placementv1beta1.PropertySelectorOperator]bool{
+		placementv1beta1.PropertySelectorGreaterThan:          true,
+		placementv1beta1.PropertySelectorGreaterThanOrEqualTo: true,
+		placementv1beta1.PropertySelectorLessThan:             true,
+		placementv1beta1.PropertySelectorLessThanOrEqualTo:    true,
+		placementv1beta1.PropertySelectorEqualTo:              true,
+		placementv1beta1.PropertySelectorNotEqualTo:           true,
+	}
+	if validOperators[op] && len(values) != 1 {
+		return fmt.Errorf("operator %s requires exactly one value, got %d", op, len(values))
+	}
+	return nil
+}
+
+func validateValues(values []string) error {
+	for _, value := range values {
+		if _, err := resource.ParseQuantity(value); err != nil {
+			return fmt.Errorf("value %s is not a valid resource.Quantity: %w", value, err)
+		}
+	}
+	return nil
+}
+
+func checkContradictions(index int, req placementv1beta1.PropertySelectorRequirement, reqs []placementv1beta1.PropertySelectorRequirement) error {
```

Comment:
Hi Britania! This part might need a bit more work. It's O(N^2) and there are some minor issues. To expedite the process I am actually OK with dropping the contradiction checks for now with a TODO comment as a hint.